### ISSUE TITLE
Tracking videos and excluding user-video combinations

### ISSUE THREAD

**Issue by bdnc** [ASL-LEX/SignLab2#94]
For videos added to a dataset via upload, it is possible to add metadata via the required CSV. Currently the metadata filename and entryID. The entryID serves to identify the video in the tag csv that can be downloaded.
It is also possible to add video responses to a dataset. In this case, it is not possible to add any metadata and the entryID appears as TODO. The video can be traced back by finding the tag csv from the original study in which it was given as a response (so you know what the cue was and which user gave it). Would it make sense to be able to add an entryID for these videos?
This is related to another topic. Users give responses in one study and then those responses are tagged in another study. Ideally, we do not want users to tag their own videos. Is there a way to block a user from seeing their own video? One way to implement this would be with an exclusion list: you could upload a list of user and entryID pairs that indicate which user-video combinations are prohibited. Another option would be to create entryIDs for the response videos that includes the contributing userID; then no video which includes a userID can be shown to that user.

**Comment by cbolles** [comment]
Yea for the metadata, we've been considering different ways on expanding the number of fields a user can provide, and hopefully allowing for editing of metadata once its been uploaded.

For the related topic, I see how that is really beneficial. We will have to discuss how best to approach this feature, I'll reach out once we have a path forward on this.

**Comment by cbolles** [comment]
I've added a potential feature to meet your need.

![image](https://github.com/ASL-LEX/SignLab2/assets/16564193/c2e9df0e-9181-4b69-a9a9-651999067710)

That checkbox on the bottom will make it so users are not shown entries that they recorded in SignLab. So the general workflow will be

1. A study (Study 1) is created where the users record videos based on some initial prompt
2. A second study (Study 2) is created where the user labels the recorded video with text, the checkbox is toggled
3. During Study 2, users won't see any of the videos they recorded in Study 1 (or really any previous entry recorded in any study)

Something to note, this change will work with any entry from this point on, so not on previously recorded entries since I had to make a tweak to how the entries are stored when users record videos.

Will this fill your need?
